quickstart/gcp-cloudshell-quickstart.py

```diff
@@ -88,20 +93,103 @@
         print('Please run "gcloud config set project [PROJECT_ID]" to set the project.')
         print('See: https://cloud.google.com/sdk/gcloud/reference/config/set')
         print('')
         sys.exit(1)
 
+    ensure_google_libraries()
+    ensure_terraform()
+
+
+def ensure_google_libraries():
+    """A function that ensures Google libraries are installed. 
+
+    The function first tries to import the required Google libraries and 
+    if the required PyPI packages are not installed, it will prompt the 
+    user to install the required packages in the user's home directory. 
+    """
+
+    # Global calls for import statements are required to avoid module not found error
+    import_google_libraries = '''\
+    import googleapiclient.discovery
+    from google.cloud import kms_v1
+    from google.cloud.kms_v1 import enums
+    from google.api_core import exceptions as google_exc
+    '''
+
+    try:
+        exec(textwrap.dedent(import_google_libraries), globals())
+        print('Successfully imported Google libraries.')
+
+    except ImportError as err:
+        required_google_libraries = 'google-api-python-client grpc-google-iam-v1 google-cloud google-cloud-kms'
+
+        install_permission = input(
+            'One or more of the following Python packages are missing:\n'
+            f'  {required_google_libraries}\n\n'
+            'The script can install these packages in the user\'s home directory using the following command:\n' 
+            f'  (i.e. {sys.executable} -m pip install {required_google_libraries} --user)\n'
+            'Proceed? (y/n)? ').strip().lower()
+
+        if install_permission not in ('y', 'yes'):
+            print('Google libraries are required for deployment. Exiting...')
+            sys.exit(1)
+
+        install_cmd = f'{sys.executable} -m pip install {required_google_libraries} --user'
+        subprocess.check_call(install_cmd.split(' '))
+
+        # Recommended to clear cache after installing python packages for dynamic imports
+        importlib.invalidate_caches()
+
+        exec(textwrap.dedent(import_google_libraries), globals())
+        print('Successfully imported Google libraries.')
+
+
+def ensure_terraform():
+    """A function that ensures Terraform version >= 0.12 is installed. 
+
+    The function first checks if Terraform is installed in the user's 
+    system and if Terraform is not installed, or it is not version >= 0.12, 
+    it will prompt the user to install Terraform in the user's home directory. 
+    """
+
+    global TERRAFORM_BIN_PATH
+
+    path = shutil.which('terraform')
+
+    if path:
+        cmd = 'terraform -v'
+        # Run the command 'terraform -v' and use the first line as the Terraform version
+        terraform_version = subprocess.run(cmd.split(' '),  stdout=subprocess.PIPE).stdout.decode('utf-8').splitlines()[0]
+        
+        print(f'Found {terraform_version} in {path}.')
+
+        if 'v0.12.' in terraform_version:
```

Comment:
this doesn't actually check for >=0.12.  For example, 0.13 would fail this check.